census_view.cpp

```diff
@@ -1317,6 +2205,62 @@ void CensusView::update_visible_columns()
         }
 }
 
+void CensusGridView::update_visible_columns()
+{
+    // Display exactly those columns whose rows aren't all identical. For
+    // this purpose, consider as "rows" the individual cells--and also the
+    // case and class defaults, even though they aren't displayed in rows.
+    // Reason: although the case and class defaults are hidden, they're
+    // still information--so if the user made them different from any cell
+    // wrt some column, we respect that conscious decision.
+    auto const& all_headers = case_parms()[0].member_names();
+    std::vector<int> new_visible_columns;
+    int column = 0;
+    for(auto const& header : all_headers)
+        {
+        if(column_value_varies_across_cells(header))
+            {
+            new_visible_columns.push_back(column);
+            }
+        ++column;
+        }
+
+    if(new_visible_columns != visible_columns_)
+        {
+        auto const selected_row = grid_window_->GetGridCursorRow();
```

Comment:
Could we use consistent names for these 2 variables, e.g. `cursor_{row,col}`?

Alternatively, we could just do `auto const cursor_coords = grid_window_->GetGridCursorCoords()`